tests/test_complex.py

```diff
@@ -104,6 +105,7 @@ def test_tool_version_delete_nodejs():
 def test_test_version_python():
     assert COMPLEX_TEXT.endswith("\n")  # Editing last item gets mangled otherwise
     stream = parse_str(COMPLEX_TEXT)
+    assert stream["options"]["test-versions"] == ["3.9"]
```

Comment:
I things parsed in-order and then being compared in-order is also at play here, it would be great to add a test showcasing that.